button-card.js

```diff
@@ -25,7 +25,7 @@ class ButtonCard extends LitElement {
 
 
   getFontColorBasedOnBackgroundColor(backgroundColor) {
-    const parsedBackgroundColor = backgroundColor.match(/^rgb\s*\(\s*(\d+)\s*,\s*(\d+)\s*,\s*(\d+)\s*\)$/i);
+    const parsedBackgroundColor = backgroundColor.match(/^rgb\s*\(\s*(\d+)\s*,\s*(\d+)\s*,\s*(\d+)\s*\)$/i) ? backgroundColor.match(/^rgb\s*\(\s*(\d+)\s*,\s*(\d+)\s*,\s*(\d+)\s*\)$/i) : this.hexToRgb(backgroundColor.substring(1));
```

Comment:
I'd do 

```javascript
const parsedRgbColor= backgroundColor.match(/^rgb\s*\(\s*(\d+)\s*,\s*(\d+)\s*,\s*(\d+)\s*\)$/i);
const parsedBackgroundColor = parsedRgbColor ? parsedRgbColor : this.hexToRgb(backgroundColor.substring(1));
```

That way you avoid going through the regex twice. It's just a simple regex but if you have 20 buttons, it adds up.